Simplenote/src/main/java/com/automattic/simplenote/utils/AuthUtils.java

```diff
@@ -27,6 +27,9 @@ public static void logOut(Simplenote application) {
         application.getTagsBucket().reset();
         application.getPreferencesBucket().reset();
 
+        // We have to verify the account for the next user
+        application.getReviewAccountVerifier().clearStatus();
```

Comment:
I see this may be a bigger problem with the design, but I don't like that we have to reset all these objects at logout. I wish instead they were destroyed and recreated when logging in. given that every other bucket is reset here I don't think it's a barrier to this PR, but wanted to ear-mark it for the future.

Reply:
I think this is the consequence of the current architecture which setups mostly everything when the application (`Simplenote`) is created. Thus, the application keeps the `state` that should be cleaned during logout.